### PR TITLE
fix(ingest) - always generated an _id, guid is going to point to the ID in the provider system

### PR DESCRIPTION
**Pull request by amagdas**
SD-1946 Fetching operation fails on recently ingested items with Post 404 not found code

This is a big change in how the system works, summary of changes:
- all the items will have an `_id` generated that is going to be different from `guid`, this is because `guid` is not always suited to be used inside URL's
- we won't be able to use `_id` and `guid` interchangeably which is a good thing IMO, we shouldn't rely on this at all, I've updated the obvious situations, hopefully I fixed all of them

@petrjasek @superdesk/developers @sivakuna-aap @akintolga @marwoodandrew @mdhaman

### CODE REVIEW

File: server/apps/duplication/archive_fetch.py [hunk 1]
```diff
@@ -52,23 +52,24 @@ class FetchResource(Resource):
...
 
-    def __fetch(self, docs, guid=None, **kwargs):
-        guid_of_fetched_items = []
+    def __fetch(self, docs, id=None, **kwargs):
+        id_of_fetched_items = []
 
         for doc in docs:
-            guid_of_item_to_be_fetched = doc['guid'] if guid is None else guid
+            id_of_item_to_be_fetched = doc.get('_id') if id is None else id
+            print('Id: ', id_of_item_to_be_fetched)
```
Comment (mugurrus) [inline server/apps/duplication/archive_fetch.py]: remove print

File: server/apps/duplication/archive_fetch.py [hunk 2]
```diff
@@ -52,23 +52,24 @@ class FetchResource(Resource):
...
+        return self.__fetch(docs, id=request.view_args.get('id'), **kwargs)
 
-    def __fetch(self, docs, guid=None, **kwargs):
-        guid_of_fetched_items = []
+    def __fetch(self, docs, id=None, **kwargs):
+        id_of_fetched_items = []
 
         for doc in docs:
-            guid_of_item_to_be_fetched = doc['guid'] if guid is None else guid
+            id_of_item_to_be_fetched = doc.get('_id') if id is None else id
```
Comment (mugurrus) [inline server/apps/duplication/archive_fetch.py]: When id is given, why do you do the ingest for all docs in the list?

Comment (sivakuna-aap) [inline server/apps/duplication/archive_fetch.py]: Agreed, but that's the syntax of service handler in `BaseService` for `POST` request

**Comment by petrjasek** [comment]
:+1: guid as _id didn't work anyhow. but I'm surprised there are no client changes..

**Comment by amagdas** [comment]
@petrjasek Will retest the client, most probably we use the self href from the entity.

File: server/apps/duplication/archive_fetch.py (hunk 2 above)
Comment (amagdas) [inline server/apps/duplication/archive_fetch.py]: @mugurrus I think the idea is to be able to apply multiple fetch actions on the same item, most probably to apply different macros for routing/publishing.

**Comment by amagdas** [comment]
@petrjasek Retested, the client side looks ok.

**Comment by amagdas** [comment]
@sivakuna-aap @akintolga @marwoodandrew @mdhaman 
Added the fix for fetching the content from celery, review please.

**Comment by mdhaman** [comment]
:+1:

**Pull request merged**